sql/expression/convert.go

```diff
@@ -166,6 +166,9 @@ func convertValue(val interface{}, castTo string, originType sql.Type) (interfac
 		}
 		return b, nil
 	case ConvertToChar, ConvertToNChar:
+		if sql.IsDecimal(originType) {
+			return val, nil
+		}
```

Comment:
I didn't fully understand why we needed this. Looks like it's short circuiting and returning the decimal type instead of converting it to a varchar/Text? Was this just an optimization or does the code not execute correctly without this?

Reply:
This is something i'm confused on, I added this to see if it makes other tests fail or not. 
This gets the correct value of decimal.Decimal type (I made change to parse code to get decimal.Decimal instead of string literal for floats that are rounded), but somewhere in analyzer the `castTo` parameter gets "char", which is causing decimal value of `2.00` to `2` which causes the decimal value to lose it's scale information. 

I'll be taking a closer look into why this is happening.